lib/mosaic-database/src/main/java/org/eclipse/mosaic/lib/database/spatial/EdgeFinder.java

```diff
@@ -53,24 +61,48 @@ public EdgeFinder(Database database) {
             }
         }
         edgeIndex = new KdTree<>(new SpatialItemAdapter.EdgeAdapter<>(), items);
-        edgeSearch = new org.eclipse.mosaic.lib.database.spatial.Edge.Nearest<>();
+        edgeSearch = new org.eclipse.mosaic.lib.database.spatial.Edge.KNearest<>();
     }
 
     /**
      * Searches for the closest edge to the geo location.
      *
      * @return Closest edge to the given location.
      */
-    public Edge findClosestEdge(GeoPoint location) {
+    public List<Edge> findClosestEdge(GeoPoint location) {
```

Comment:
Instead of returning a list here, would it be sufficient to pass another parameter `heading` and then choose the edge which is closest to the heading direction?

Reply:
otherwise, the method should be renamed to "findClosestEdges` and needs an updated java doc

Reply:
- Do you mean to add this as an overload?
- Oh yes I totally missed the documentation

Reply:
I just don't want a method that is called "findClosestEdge" which returns a list. Therefore you could indeed offer both methods (`findClosestEdge(location, heading)` and `findClosestEdges(location)`).